server/server.js

```diff
@@ -0,0 +1,59 @@
+import express from "express";
+import axios from "axios";
+import dotenv from "dotenv";
+import cors from "cors";
+
+dotenv.config();
+
+const app = express();
+const PORT = 8080;
+
+app.use(
+  cors({
+    origin: process.env.URL || "http://localhost:5173",
+    allowedHeaders: "Origin, X-Requested-With, Content-Type, Accept",
+  })
+);
+
+app.get("/weather", async (req, res) => {
+  try {
```

Comment:
A nitpick - usually we will start wrapping code in a `try` block immediately above the line that could throw the error. In your case, `await axios.get()` is what might throw an error, so it's better practice to start the `try` block immediately above this line.